### PR TITLE
COMMERCE-9383 As a Developer I want to implement InfoItemRenderer for OptionSelector

### PR DESCRIPTION
**Pull request by alessiorendina**
https://issues.liferay.com/browse/COMMERCE-9383

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:stop

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java [hunk 1]
```diff
@@ -91,17 +110,67 @@ public Sku toDTO(DTOConverterContext dtoConverterContext) throws Exception {
...
+				incomingQuantityLabel =
+					_cpContentHelper.getIncomingQuantityLabel(
+						cpSkuDTOConverterConvertContext.getCompanyId(),
+						cpSkuDTOConverterConvertContext.getLocale(),
+						cpInstance.getSku(),
+						cpSkuDTOConverterConvertContext.getUser());
 				manufacturerPartNumber = cpInstance.getManufacturerPartNumber();
 				price = _getPrice(
-					cpInstance, 1,
+					cpInstance, cpSkuDTOConverterConvertContext.getQuantity(),
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java:121]: What if the quantity is not 1 in the quantity selector? How do you manage that?

File: modules/apps/commerce/commerce-product-ddm-impl/src/main/java/com/liferay/commerce/product/ddm/internal/DDMHelperImpl.java [hunk 2]
```diff
@@ -470,20 +475,25 @@ private boolean _isIterableCPDefinitionOptionRelFieldType(
...
-			RenderResponse renderResponse)
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws PortalException {
+
+		return _render(
+			cpDefinitionId, locale, ddmForm, json, null, httpServletRequest,
+			httpServletResponse);
+	}
+
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-ddm-impl/src/main/java/com/liferay/commerce/product/ddm/internal/DDMHelperImpl.java:486]: Why do you need this, just call the other one passing null for page context

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/info/item/renderer/OptionSelectorInfoItemRenderer.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	public void render(
+		CPDefinition cpDefinition, HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse) {
+
+		if (cpDefinition == null) {
+			return;
+		}
+
+		try {
+			CommerceContext commerceContext =
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/info/item/renderer/OptionSelectorInfoItemRenderer.java:70]: what if you put this fragment outside of Commerce?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Ehi @alessio-rendina, please have a look at the comments that I left and let me know what do you think!

Thank you!

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java (hunk 1 above)
Comment (alessiorendina) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java:121]: The quantity is a queryParameter, so we can pass it in query after reading it from quantity selector

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/info/item/renderer/OptionSelectorInfoItemRenderer.java (hunk 3 above)
Comment (alessiorendina) [reply modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/info/item/renderer/OptionSelectorInfoItemRenderer.java:70]: This fragment can be used only for CPDefinition

Comment (alessiorendina) [reply modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/info/item/renderer/OptionSelectorInfoItemRenderer.java:70]: `implements InfoItemRenderer<CPDefinition>`

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/info/item/renderer/OptionSelectorInfoItemRenderer.java:70]: got it, right! now I see, thank you!

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java (hunk 1 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java:121]: Let's see this together I want to make sure I'm on the same page with you

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by 22LucaGiannotta22** [comment]
ci:stop

**Comment by 22LucaGiannotta22** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Pull request closed without merging**